src/qtiCommonRenderer/renderers/interactions/pci/ims.js

```diff
@@ -72,8 +97,19 @@ export default function defaultPciRenderer(runtime) {
         destroy: function destroy(interaction) {
             instanciator.getPci(interaction).oncompleted();
         },
+        /**
+         * IMS PCI does not have setState, so PCI should be destroyed and reinstanciated with response.
+         * This function should run only in review mode.
+         * @param {Object} interaction
+         * @param {Object} state - state that should be set
+         * @returns {Promise<Object>} - Resolves with newly created instance
+         */
+        setReviewState(interaction, state) {
+            this.destroy(interaction);
+            return this.createInstance(interaction, { response: { RESPONSE: state.response } });
```

Comment:
Based on my investigation, it is not necessary for review mode. I tried to set a unique state, but when system calls `setState` on Result page, it restores only the response. (I do not know in this case why is it called `setState`). So, this is the reason why I convert the "state" to response. Maybe you saw, I always use the `RESPONSE` as response identifier, because I do not know what was the original response identifier, but it should not cause an issue, because the interaction is in review mode, so we won't save its response.
BTW, in nextgen we do the same, so we restore only the response in review mode. I think it is good, because some interaction saves half answers (for example associate interaction) in state, that wouldn't be good to be restored.
I checked the the IMS PCI doc and I found: "The state parameter is optional. It can be passed to recreate a previous interaction state of the item."

Reply:
Ok fine for the state.
However, I didn't pay attention to the `RESPONSE` identifier, which may cause issues if the PCI is used aside other interactions. Could you see if you can retrieve the correct identifier from the response declaration?

Reply:
Hmm, could you tell me how could it cause issue? The response identifier is passed into the PCI in `boundTo` and thats all. How could it cause an issue, if another interaction would have the same identifier? The response won't be requested (and sent) because it review.